UsageEnvironment/include/Log.hh

```diff
@@ -16,20 +24,48 @@ enum LogLevel {
 
 class _Log {
 private:
-  char      logFilePath[256];
+  const unsigned int kMaxLogs = 6;
+ 
+  char      logFilePath[PATH_MAX];
+  unsigned int logFileNum;
   bool      logToStdout;
   LogLevel  logLevel;
 
   FILE* AcquireFile();
   void  ReleaseFile(FILE* logFileHandle);
+  int   NextLogFile();
+  int   PreviousLogFile();
+  int   SetLogFileNum(unsigned int logNum);
+  void  RollLogs();
 public:
   _Log() : logToStdout(false), logLevel(LogLevelDebug) {
-    strcpy(logFilePath, "/mnt/ramdisk/live555.log");
-  }
+# ifdef LOG_FILE_TMP_DIR
+    if (kMaxLogs == 0) {  // Use stdout
+      strcpy(logFilePath, "");
+      logToStdout = true;
+      return;
+    }
+
+    // Set log file to 0
+    SetLogFileNum(0);
 
-  ~_Log() {
+    // If log files already exist, find the last one then continue logging on top of them
+    int errNext = -1;
+    while ((access(logFilePath, F_OK) == 0) && ((errNext = NextLogFile()) == 0)) {}
+    if (errNext == 0) {
+      PreviousLogFile();
+    }
+
+# else
+  // Use stdout
+  strcpy(logFilePath, "");
+  logToStdout = true;
+# endif
```

Comment:
Any function that's more than a single line (or two at the most) should be moved to the cpp file.

Reply:
+1️⃣